devito/propagator.py

```diff
@@ -200,7 +201,7 @@ def sympy_to_cgen(self, stencils):
                 # TODO: add support for double precision
                 self.add_local_var(name, np.float32)
                 # TODO: undo precision enforcing
-                factors.append(cgen.Assign(name, str(ccode(expr.xreplace(self._var_map))).replace("pow", "powf").replace("fabs", "fabsf")))
+                factors.append(cgen.Assign(name, str(ccode(self.time_substitutions(expr).xreplace(self._var_map))).replace("pow", "powf").replace("fabs", "fabsf")))
```

Comment:
TLDR: I have expressed my reservations. I am not stopping the merge. Go for it @ggorman 

This is primarily because there were some assumptions in the code that weren't expressively coded as tests. This was due to the priorities at the time I was writing the code (as I understood them from my communication with you @ggorman ). I would even dare to say that tests were less important in the recent past when it was primarily one person working on the code base. When the team grew, the focus should have been on tests - and @mlange05 and I were (and are) trying to enforce it to the extent possible.  That green is only as good as our tests and its important to  remember that our tests are a serious work in progress.   

Whatever I am doing is in the interest of maintaining stability in the codebase in the future. Admittedly, this change does not do enough damage to warrant holding on to this PR. It is, however, a (tiny) step in the direction that I have discovered to be wrong. In working towards implementing aligned memory allocation directly in C on the `first_touch` branch, I am using the low-level Propagator interface that has been opened up with the last few PRs and I see that this will change completely when I am done there. Just trying to point it out here so people are aware of what's coming up. It is fine to let this go into master and fix it after my PR lands.